src/app/settings/admin/presets/dialog/steps/settings/provider/gcp/component.ts

```diff
@@ -74,13 +74,19 @@ export class GCPSettingsComponent extends BaseFormValidator implements OnInit, O
   }
 
   private _update(): void {
-    const serviceAccountValue = isValid(this.form.get(Controls.ServiceAccount).value)
-      ? this.form.get(Controls.ServiceAccount).value
-      : encode(this.form.get(Controls.ServiceAccount).value);
     this._presetDialogService.preset.spec.gcp = {
-      serviceAccount: serviceAccountValue,
+      serviceAccount: this._serviceAccountValue,
       network: this.form.get(Controls.Network).value,
       subnetwork: this.form.get(Controls.Subnetwork).value,
     } as GCPPresetSpec;
   }
+
+  private get _serviceAccountValue(): string {
+    let serviceAccountValue = this.form.get(Controls.ServiceAccount).value;
+    if (!!serviceAccountValue && !isValid(this.form.get(Controls.ServiceAccount).value)) {
+      serviceAccountValue = encode(this.form.get(Controls.ServiceAccount).value);
+    }
```

Comment:
Is there a reason why the variable is assigned at the first line and then it is not being really used in the `if` and during the `encode`.

Reply:
if Monday counts... otherwise no. Fixed.